hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,311 @@
+
+        GPUArray<unsigned int> m_bounced_index;            //!< Indices of bounced particles
+        GPUVector<unsigned int> m_picks;                   //!< Particles picked for evaporation on this rank
+        GPUVector<mpcd::detail::pdata_element> m_removed;  //!< Hold output particles that are removed
+
+        virtual void applyPicks();                         //!< Apply the picks
+
+    private:
+        std::vector<unsigned int> m_all_picks;             //!< All picked particles on all the ranks
+
+        unsigned int calculateNumbounced();                //!< For calculating N_bounced and m_bounced_index 
+        //!< For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick, unsigned int N_bounced_total);
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    // use peekStream since shouldStream will be called by parent class 
+    if(!peekStream(timestep)) return;
+    
+    // compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_mpcd_dt);
+    // checks if V <= 0, since size of droplet must decrease
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (!m_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_validate_geom = false;
+        }
+
+    /*
+     *Update the geometry radius to the size at the end of the streaming step.
+     * This needs to be done every time.
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to evaporate(such that solvent density remain constant)
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const unsigned int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
+
+    // get the compact array of indexes of bounced particles and total N_bounced
+    unsigned int N_bounced = calculateNumbounced();
+
+    // reduce / scan the number of particles that were bounced on all ranks
+    unsigned int N_bounced_total = N_bounced;
+    unsigned int N_before = 0;
+    #ifdef ENABLE_MPI
+    if (m_exec_conf->getNRanks() > 1)
+        {
+        MPI_Allreduce(&N_bounced, &N_bounced_total, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
```

Comment:
This can just be in-place because you have initialized it above, it makes the intent clearer:
```suggestion
        MPI_Allreduce(MPI_IN_PLACE, &N_bounced_total, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
```